### PR TITLE
docs: Update cloudformation.yaml docs for v1

### PR DESCRIPTION
**Pull request by chrisnegus**
**Description**: This PR updates the [Cloudformation](https://karpenter.sh/docs/reference/cloudformation/) docs to match the new cloudformation.yaml for v1.

**Does this change impact docs?**
- [X ] Yes, PR includes docs updates 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/preview/reference/cloudformation.md [hunk 1]
```diff
@@ -390,6 +396,7 @@ Also, `karpenter.k8s.aws/ec2nodeclass` must be set to some value. This ensures t
   "Condition": {
     "StringEquals": {
       "aws:RequestTag/kubernetes.io/cluster/${ClusterName}": "owned",
+      "aws:RequestTag/eks:eks-cluster-name": "${ClusterName}",
```
Comment (jonathan-innis) [inline website/content/en/preview/reference/cloudformation.md:399]: Do we need to make an update to the section above to include the fact that we are now requiring that the CloudFormation set the `eks:eks-cluster-name` tag in the request?

File: website/content/en/preview/reference/cloudformation.md [hunk 2]
```diff
@@ -213,6 +214,7 @@ actions on `fleet`, `instance`, `volume`, `network-interface`, `launch-template`
   "Condition": {
     "StringEquals": {
       "aws:RequestTag/kubernetes.io/cluster/${ClusterName}": "owned",
+      "aws:RequestTag/eks:eks-cluster-name": "${ClusterName}"
```
Comment (jonathan-innis) [inline website/content/en/preview/reference/cloudformation.md:217]: Same comments here. Does this change anything about what we specify in the sections that we have above?

File: website/content/en/preview/reference/cloudformation.md (hunk 1 above)
Comment (chrisnegus) [reply website/content/en/preview/reference/cloudformation.md:399]: Done.

**Review by jonathan-innis — DISMISSED** [review]
LGTM 🚀

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**